### PR TITLE
Chat UI

### PR DESCRIPTION
**Pull request by Boukhari-Mohamed**
Add tests for chat UI.
Refractor Models and Adapters.
Add a high level layout for the background.

### CODE REVIEW

**Review by khalilacheche — APPROVED** [review]
Nothing to declare

File: app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/adapters/MessageListAdapter.java [hunk 1]
```diff
@@ -115,13 +121,12 @@ void bind(ChatMessageModel message) {
             profileImage = (ImageView) itemView.findViewById(R.id.image_gchat_profile_other);
         }
 
-        void bind( ChatMessageModel message) {
-            messageText.setText(message.getMessageText());
-
+        void bind( Message message) {
```
Comment (Roilek) [inline app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/adapters/MessageListAdapter.java:124]:
```suggestion
        void bind(Message message) {
```

File: app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/adapters/MessageListAdapter.java [hunk 2]
```diff
@@ -21,27 +21,33 @@
...
     public int getItemViewType(int position) {
-        ChatMessageModel message = (ChatMessageModel) mMessageList.get(position);
+        Message message = (Message) mMessageList.get(position);
 
-        //if (message.getSender().getUserId().equals(SendBird.getCurrentUser().getUserId())) {
-        //this needs to be revised and corrected to adhere to the userProfileModel and message model
-        //this code is here for demo purposes
-        if (message.getMessageUser().equals("Elliot")){
+        if (message.getSender().getId() == AuthenticatorFactory.getDependency()
+                .getCurrentUser().getId()){
```
Comment (Roilek) [inline app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/adapters/MessageListAdapter.java:50]:
```suggestion
                .getCurrentUser().getId()) {
```

File: app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/ui/activities/ChatRoomActivity.java [hunk 3]
```diff
@@ -13,45 +12,50 @@
...
+
+        //how are we gonna edit this
+        DummyDatabase dep = DummyDatabase.getInstance();
+        messageList = new ArrayList<>(dep.fetchMessages());
+
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_chat_room);
 
-
+        //chatter = (User) getIntent().getSerializableExtra("MyClass");
```
Comment (Roilek) [inline app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/ui/activities/ChatRoomActivity.java:43]: Should not keep this commented out

File: app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/ui/activities/ChatRoomActivity.java [hunk 4]
```diff
@@ -13,45 +12,50 @@
...
 
         //back arrow
         Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar_gchannel);
         setSupportActionBar(toolbar);
 
         // add back arrow to toolbar and remove title
         if (getSupportActionBar() != null){
```
Comment (Roilek) [inline app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/ui/activities/ChatRoomActivity.java:50]:
```suggestion
        if (getSupportActionBar() != null) {
```

**Review by Roilek — APPROVED** [review]
Some light space refactoring.
Careful to not merge not clean debug lines or // lines that are not meant to exist or notre even a comment

File: app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/adapters/MessageListAdapter.java [hunk 5]
```diff
@@ -21,27 +21,33 @@
...
+    public MessageListAdapter(Context context, List<Message> messageList) {
         mContext = context;
         mMessageList = messageList;
+
+    }
+
+
+    public void addMessage(Message message){
+        mMessageList.add(message);
+        this.notifyDataSetChanged();
```
[the remark this answers is not in the document]
Comment (Boukhari-Mohamed) [reply app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/adapters/MessageListAdapter.java:35]: notifyDataSetChanged() is supposed to notify the RecyclerView when the list it displays changes which is exactly the case here: Whenever we add a message to the conversation(List of messages) we notify the View.

**Review by Boukhari-Mohamed — COMMENTED** [review]
approved

**Pull request merged**